integration/spark/src/main/scala/org/apache/carbondata/spark/load/DataLoadProcessBuilderOnSpark.scala

```diff
@@ -227,9 +232,15 @@ object DataLoadProcessBuilderOnSpark {
     // 2. sort
     var numPartitions = CarbonDataProcessorUtil.getGlobalSortPartitions(
       configuration.getDataLoadProperty(CarbonCommonConstants.LOAD_GLOBAL_SORT_PARTITIONS))
+
+    // if numPartitions user does not specify and not specified in config then dynamically calculate
     if (numPartitions <= 0) {
-      numPartitions = originRDD.partitions.length
+      val defaultMaxSplitBytes = sessionState(sparkSession).conf.filesMaxPartitionBytes
+      val dynamicPartitionNum = Math.ceil(SizeEstimator.estimate(originRDD).toDouble /
```

Comment:
does SizeEstimator.estimate work for RDD?

Reply:
actually if you see the `SizeEstimator.estimate`, it takes any object and estimate the number of bytes that the object takes up on JVM heap.

and RDD actually contain the logical batches/logical partitions of data right, so is this calculation correct? correct me if i am wrong.

@QiangCai

Reply:
@akashrn5 right, it isn't the data size.